functional_tests/android_test/tests.py

```diff
@@ -0,0 +1,139 @@
+
+	def testGccArm64CompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=arm64", "--toolchain=android-gcc"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testGccMipsCompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=mips", "--toolchain=android-gcc"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testGccMips64CompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=mips64", "--toolchain=android-gcc"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testClangX86CompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=x86", "--toolchain=android-clang"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testClangX64CompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=x64", "--toolchain=android-clang"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testClangArmCompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=arm", "--toolchain=android-clang"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testClangArm64CompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=arm64", "--toolchain=android-clang"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testClangMipsCompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=mips", "--toolchain=android-clang"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
+
+	def testClangMips64CompileSucceeds(self):
+		"""Test that the project succesfully compiles"""
+		testArgs = ["--project=hello_world", "--arch=mips64", "--toolchain=android-clang"]
+		self.cleanArgs = testArgs
+		self.assertMakeSucceeds("-v", "--show-commands", *testArgs)
+
+		self.assertTrue(os.access(self.outputFile, os.F_OK))
```

Comment:
Wish there were a way to load these compiled APKs up into the emulator and verify they work as part of the test. That sounds like it would be extremely complicated though.

Reply:
Pretty sure it's possible, but yes, it would be very complicated.  Also, we have no APK packaging right now, so not something we could implement at the moment anyway.